### DISCUSSION TITLE
Programmatically get access to other drivers/sensors

### DISCUSSION THREAD

**Discussion by Heizelmann** [arendst/Tasmota#18306] [Q&A]
E.g. from my own driver modul I would like to access the temperature value of a ds18x20.

**Comment by sfromis** [comment]
Simplest way of doing that would be to code your driver in Berry on ESP32, where you can access all sensor data easily.

For C++ code, an existing temperature sharing mechanism, used by some drivers, is that temperature sensors should be updating `TasmotaGlobal.temperature_celsius`, available elsewhere,  avoiding the dependence on what specific sensor it happens to be.

**Comment by Heizelmann** [comment]
Thanks for the reply, but this is not a solution for me. I am using only ESP hardware and the `xsns_05_ds18x20` sensor I am using  does not use `TasmotaGlobal.temperature_celsius`

**Reply by sfromis** [reply]
That's not correct. It calls `ConvertTemp` which calls `UpdateGlobalTemperature` which unsurprisingly is where `TasmotaGlobal.temperature_celsius` is updated.

**Reply by Heizelmann** [reply]
Thanks. I didn't saw this.

**Comment by Heizelmann** [comment]
I have now the same problem with analog sensors, e.g. a simple lumminance sensor which I would like to access programmatically from my own driver module.  I can not find that this sensor has a connection to TasmotaGlobal.
see #18328

**Reply by arendst** [reply]
If you need access to other connected sensors I suggest you take a look at the prometheus driver (xsns_75_prometheus.ino) how they manage to pick up sensor data WITHOUT a total rewrite of sensor drivers.
